website/blog/2022-12-13-pointer-events-in-react-native.md

```diff
@@ -0,0 +1,175 @@
+
+Using Android as an example, the general approach to leveraging platform events are:
+
+1. Iterate through all pointers of the `MotionEvent` and do a depth-first search to determine the target React view of each pointer and its ancestral path.
+2. Map the category of `MotionEvent` to the relevant pointer events. There is a 1-to-many relationship between `MotionEvent` and `PointerEvent`. In the illustration of their relationship, dotted lines indicate fired events if the pointing device does not support hover.
+
+![A diagram illustrating the relationship of types of Android MotionEvents into Pointer Events fired. Some pointer events are conditionally fired if pointing device does not support hover. “ACTION_DOWN” and “ACTION_POINTER_DOWN” fire pointerdown and conditionally fire pointerenter, pointerover. “ACTION_MOVE” and “ACTION_HOVER_MOVE” fire pointerover, pointermove, pointerout, pointerup. “ACTION_UP” and “ACTION_POINTER_UP” fire pointerup and conditionally fire pointerout, pointerleave.](/blog/assets/pointer-events-motionevent-relationship.png)
+
+1. Build the `PointerEvent` interface with platform details from the `MotionEvent` and cached state of previous interactions. (Ex. [the `button` property](https://w3c.github.io/pointerevents/#the-button-property))
+2. Dispatch the pointer events from Android to React Native’s [core event queue](https://github.com/facebook/react-native/blob/main/ReactCommon/react/renderer/core/EventQueueProcessor.cpp#L20) and leverage JSI to call the [`dispatchEvent`](https://github.com/facebook/react/blob/main/packages/react-native-renderer/src/ReactFabricEventEmitter.js#L83) method in `react-native-renderer` which iterates through the React tree for the bubble and capture phase of the event.
+
+## Implementation Progress
+
+When it comes to our current progress of implementing the Pointer Events specification we’ve focused on a solid baseline implementation of the most common events that handle things like pressing, hovering, and moving.
+
+### Events
+
+| Implemented    | Work in Progress  | Yet to be Implemented |
+| -------------- | ----------------- | --------------------- |
+| onPointerOver  | onPointerCancel\* | onClick               |
+| onPointerEnter |                   | onContextMenu         |
+| onPointerDown  |                   | onGotPointerCapture   |
+| onPointerMove  |                   | onLostPointerCapture  |
+| onPointerUp    |                   | onPointerRawUpdate    |
+| onPointerOut   |                   |                       |
+| onPointerLeave |                   |                       |
+
+<!-- prettier-ignore -->
+<sub>* onPointerCancel has been hooked up to the native platform’s “cancel” event but this does not necessarily correspond to when the web platform expects them to fire.</sub>
```

Comment:
It would be nice to replace this note with admonition (https://docusaurus.io/docs/next/markdown-features/admonitions) fot the consistency, since `<sub>` elements are not used anywhere else on the website.